src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ComEventInterfaceAttribute.cs

```diff
@@ -1,17 +1,20 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics.CodeAnalysis;
+
 namespace System.Runtime.InteropServices
 {
     [AttributeUsage(AttributeTargets.Interface, Inherited = false)]
     public sealed class ComEventInterfaceAttribute : Attribute
     {
-        public ComEventInterfaceAttribute(Type SourceInterface, Type EventProvider)
+        public ComEventInterfaceAttribute([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)] Type SourceInterface, Type EventProvider)
```

Comment:
I think we will want this for the `EventProvider` argument as well. Same for the property below.

Reply:
I looked and couldn't find any usages of the EventProvider property. Do you have any examples of how this is used?

Would we also need PublicConstructors preserved? Or how else would the EventProvider get instantiated?

Reply:
This is used internally and honestly it is COM so everything that is involved on this type should be preserved in all manners. As soon as the linker finds any COM related type that is consuming a managed type, that managed type should be considered hands off. An example of how this attribute is use is in our tests - https://github.com/dotnet/runtime/blob/master/src/tests/Interop/COM/ServerContracts/Server.Events.cs. The attribute itself is typically only used by the generated PIA from `TlbImp`. Runtime usage is at https://github.com/dotnet/runtime/blob/cc1198a1de867915ef0c604e4426e708c0b9c9b2/src/coreclr/vm/class.cpp#L1834-L1855.

I cannot stress what I said above enough - "As soon as the linker finds any COM related type that is consuming a managed type, that managed type should be considered completely hands off.". Whenever the linker tries to work with COM it utterly fails and creates very complicated and inscrutable failures.

Reply:
Thanks for the catch here, @AaronRobinsonMSFT. I've annotated the Type with `PublicCtors | NonPublicCtors | Public Methods|Events|Fields|Properties`.

This allows for one more ILLink warning to be removed in CoreLib, so I resolved that one as well.